gwnr/waveform/esigma_utils.py

```diff
@@ -468,83 +675,132 @@ def get_imr_esigma_modes(
-        retval            -- Hybridization related data.
-                             Returned only if "return_hybridization_info" is True
+                             Returned only if the flag `return_orbital_params`
+                             is set
+        retval            -- Hybridization related data. Returned only if the
+                             flag `return_hybridization_info` is set
     """
     if not hasattr(ls, merger_ringdown_approximant):
         raise IOError(
             """We cannot generate individual modes for {merger_ringdown_approximant}.
                       Try one of: [NRSur7dq4, SEOBNRv4PHM]"""
         )
+    if (mean_anomaly is None) and (coa_phase is None):
+        raise IOError(
+            f"""Please specify one of the phase angles, either of
+                      `mean_anomaly` or `coa_phase`."""
+        )
+    if hybridize_aligning_merger_to_inspiral and (coa_phase is None):
+        raise IOError(
+            f"""If you want to attach ESIGMA inspiral to merger, by
+                      phase shifting merger to inspiral, please specify the
+                      phase angle `coa_phase`"""
+        )
+    if (not hybridize_aligning_merger_to_inspiral) and (mean_anomaly is None):
+        raise IOError(
+            f"""If you want to attach ESIGMA inspiral to merger, by
+                      phase shifting inspiral to merger, please specify the
+                      phase angle `mean_anomaly`"""
+        )
+    if mean_anomaly is None:
+        mean_anomaly = 0
+    if coa_phase is None:
+        coa_phase = 0
 
-    if return_orbital_params is True:
+    if return_orbital_params == True:
         return_orbital_params = ["x", "e", "l", "phi", "phidot", "r", "rdot"]
-
-    if isinstance(return_orbital_params, list):
+    elif isinstance(return_orbital_params, list):
         return_orbital_params_user = return_orbital_params.copy()
-    else:
+        return_orbital_params_user = set(return_orbital_params_user).intersection(
+            set(["x", "e", "l", "phi", "phidot", "r", "rdot"])
+        )
+    elif not return_orbital_params:
+        return_orbital_params = []
         return_orbital_params_user = False
 
-    sample_rate = int(np.round(1.0 / delta_t))
-
-    # If the user does not provide the 22-mode frequency at which the inspiral
-    # should transition to merger-ringdown, we use sensible defaults here.
-    if f_mr_transition is None:
-        # Kerr ISCO frequency
-        f_Kerr = f_ISCO_spin(mass1, mass2, spin1z, spin2z)
-        # Schwarzschild ISCO frequency
-        f_Schwarz = 6.0**-1.5 / (mass1 + mass2) / lal.MTSUN_SI / lal.PI
-        f_mr_transition = min(f_Kerr, f_Schwarz)
-
-    if not return_orbital_params:
-        return_orbital_params = ["e"]
-        # Needed necessarily for hybridization error printing
     return_orbital_params = set(return_orbital_params)
+    return_orbital_params_user = set(return_orbital_params)
```

Comment:
Did you mean to do `return_orbital_params_user = set(return_orbital_params_user)` here?

Reply:
Not really, in fact this line is redundant and I've removed it. `return_orbital_params_user` basically tracks the parameters to be returned to user, excluding any params we might need for internal purposes. Move this line in the `if..elif..elif` clause above.